### PR TITLE
tests

### PR DESCRIPTION
**Pull request by Farnazmdi**
I added a test in `testHill.jl` at the very last line to test the gradient function for a set of parameters in which the nD1 is zero, and it fails.

Apart from this, for now, just Gemcitabine doesn't behave expectedly in terms of the gradient. I guess the notebooks should run now.

### CODE REVIEW

**Comment by aarmey** [comment]
Should be fixed now. finite_difference defaults to central differencing, which moves your current point up and down slightly. You can't go down if you're at 0. I switched it to forward differencing.

**Comment by Farnazmdi** [comment]
Thank you very much.

**Comment by Farnazmdi** [comment]
@aarmey should we merge this?

File: test/testHill.jl [hunk 1]
```diff
@@ -12,4 +13,7 @@
 
     # Check that these at least run
     effects = getODEparams(params, conc)
+    eff2 = getODEparams(p2, conc)
+    num = numcells(effects[:, 3], g1[1]+g2[1], 100)
```
Comment (aarmey) [inline test/testHill.jl]: Can you put this into a test set with a separate label? You’re specifically testing that this works with nd2 = 0, which is a different goal than the other testing code here.

File: src/Hill.jl [hunk 2]
```diff
@@ -149,7 +152,7 @@ end
 function diffCell(params, g0, T)
     diffcells(x) = numcells(x, g0, T)
 
-    return Calculus.finite_difference(diffcells, params) / diffcells(params)
+    return Calculus.finite_difference(diffcells, params, :forward) / abs(diffcells(params))
```
Comment (aarmey) [inline src/Hill.jl]: Don’t need abs() as values are always positive.

File: src/Hill.jl [hunk 3]
```diff
@@ -138,9 +138,12 @@ end
 
 """ Calculate the # of cells in G1 for a set of parameters and T """
 function numcells(params, g0, T)
-    t = LinRange(0.0, 95.5, 192)
+    @assert(all(params .>= 0.0), "negative params $params")
+    t = LinRange(0.0, 200.0, 201)
     G1, G2 = predict(params, g0, t, Int(floor(params[6])), Int(floor(params[7])), Int(floor(params[8])), Int(floor(params[9])))
 
+    @assert(all(x -> x>=0.0, G1[2:end]), "negative cell number in G1 $G1")
```
Comment (aarmey) [inline src/Hill.jl]: White vectorized assertion like params one above.

**Pull request merged**